src/heap_lazy.rs

```diff
@@ -17,18 +18,24 @@ impl<T: Sync> Lazy<T> {
     pub const INIT: Self = Lazy(0 as *const T, ONCE_INIT);
 
     #[inline(always)]
-    pub fn get<F>(&'static mut self, f: F) -> &T
+    pub fn get<F>(&'static self, f: F) -> &T
     where
         F: FnOnce() -> T,
     {
-        unsafe {
-            let r = &mut self.0;
-            self.1.call_once(|| {
-                *r = Box::into_raw(Box::new(f()));
-            });
-
-            &*self.0
-        }
+        self.1.call_once(|| {
+            let r = Box::into_raw(Box::new(f()));
+            
+            // This is safe because the `Once` guarantees we
+            // have exclusive access to the value field
+            // The value hasn't been previously initialized
+            unsafe {
+                ptr::write(&self.0 as *const _ as *mut _, r);
```

Comment:
You can't turn the `&'static self` into a mutable pointer without `UnsafeCell`.

Reply:
I thought that might be the case. I'll revert this last commit and bump our minimum version to `1.22` so that we can use `Cell` properly.

We might have to be a bit patient with rolling this one out, the last version bump we did for `1.1.0` was a bit unpopular (to be fair that bumped about 10 versions, instead of 1).